recipes/minizip-ng/all/conanfile.py

```diff
@@ -0,0 +1,171 @@
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+        if self.options.mz_compatibility:
+            self.provides = "minizip"
+        if not tools.is_apple_os(self.settings.os):
+            del self.options.with_libcomp
+        elif self.options.with_libcomp:
+            del self.options.with_zlib
+
+    def requirements(self):
+        if self.options.get_safe("with_zlib"):
+            self.requires("zlib/1.2.11")
+        if self.options.with_bzip2:
+            self.requires("bzip2/1.0.8")
+        if self.options.with_lzma:
+            self.requires("xz_utils/5.2.5")
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.8")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1i")
+        if self.settings.os != "Windows":
+            if self.options.with_iconv:
+                self.requires("libiconv/1.16")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "minizip-ng-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["MZ_FETCH_LIBS"] = False
+        self._cmake.definitions["MZ_COMPAT"] = self.options.mz_compatibility
+        self._cmake.definitions["MZ_ZLIB"] = self.options.get_safe("with_zlib", False)
+        self._cmake.definitions["MZ_BZIP2"] = self.options.with_bzip2
+        self._cmake.definitions["MZ_LZMA"] = self.options.with_lzma
+        self._cmake.definitions["MZ_ZSTD"] = self.options.with_zstd
+        self._cmake.definitions["MZ_OPENSSL"] = self.options.with_openssl
+        self._cmake.definitions["MZ_LIBCOMP"] = self.options.get_safe("with_libcomp", False)
+
+        if self.settings.os != "Windows":
+            self._cmake.definitions["MZ_ICONV"] = self.options.with_iconv
+            self._cmake.definitions["MZ_LIBBSD"] = self.options.with_libbsd
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "BZIP2_FOUND",
+                              "BZip2_FOUND")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "BZIP2_VERSION_STRING",
+                              "BZip2_VERSION")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "BZIP2_INCLUDE_DIRS",
+                              "BZip2_INCLUDE_DIRS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "BZIP2_LIBRARIES",
+                              "BZip2_LIBRARIES")
```

Comment:
not required anymore, bzip2 recipe now provides those variables in `cmake_find_package` generator.

Reply:
done